### DISCUSSION TITLE
Problem with C++ and jdbc

### DISCUSSION THREAD

**Discussion by 5h4d0v** [microsoft/vscode-cpptools#11170] [General]
Hi, I made phonebook program, I don't have any syntax error, but when I try to compile it in VSCode I get this error message and my program doesn't start: 
<details><summary>Error Message</summary>
<p>

C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\Jakub\AppData\Local\Temp\ccdsJOYi.o:main.cpp:(.text+0x1f): undefined reference to `check(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\Jakub\AppData\Local\Temp\ccdsJOYi.o:main.cpp:(.text+0x53): undefined reference to `check(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\Jakub\AppData\Local\Temp\ccdsJOYi.o:main.cpp:(.text+0xba): undefined reference to `sql::mysql::_get_driver_instance_by_name(char const*)'
collect2.exe: error: ld returned 1 exit status

</p>
</details>  

How can I fix this issue? Thanks for help in advance

**Comment by sean-mcmanus** [comment]
It sounds like you may need to explicitly instantiate the template so that the definition gets generated from the template. You can search online for that. Maybe https://stackoverflow.com/questions/4933056/how-do-i-explicitly-instantiate-a-template-function .
